### PR TITLE
Refactor CSV Backend and add unit tests

### PR DESCRIPTION
**Pull request by josemauro**
This issue simplify code of CSV Backend switching from pandas to python csv 
module to manage csv files. This fix issue #12 improving changing the methods 
save, get and delete. Besides that, the internal method _put_value() was 
removed once it was used add lines in pandas data frames.
Methods to test successful operations to save, get and delete were added to 
automatize tests of CSV Backend.

### CODE REVIEW

File: backends/csvbackend.py [hunk 1]
```diff
@@ -68,52 +64,60 @@ def _read_config(self, settings):
...
-            store_value.to_csv(f_name, sep=',', header=True,
-                               mode='w', index=False)
-
-    def delete(self, file, start=None, end=None):
-        """ Delete a instances of the csv file"""
-        dataframe, file = self._load_file(file)
-        search = _make_search(start, end, dataframe)
-        data_to_drop = dataframe[search]
-        dataframe.drop(data_to_drop.index, inplace=True)
+            csvfile = open(fname, 'w', newline='')
```
Comment (hdiogenes) [inline backends/csvbackend.py]: You can always open with `mode='a'`.

File: backends/csvbackend.py [hunk 2]
```diff
@@ -68,52 +64,60 @@ def _read_config(self, settings):
...
+        fname = f"{self.user}_{namespace}.csv"
         if self.path != '/':
-            f_name = self.path + f_name
-
-        if f_name in os.listdir(self.path):
-            store_value.to_csv(f_name, sep=',', header=False,
-                               mode='a', index=False)
+            fname = f"{self.path}{fname}"
+ 
+        if fname in os.listdir(self.path):
```
Comment (hdiogenes) [inline backends/csvbackend.py]: Use `os.path.exists`

File: backends/csvbackend.py [hunk 3]
```diff
@@ -68,52 +64,60 @@ def _read_config(self, settings):
...
 
-        f_name = self.user + '_' + namespace + '.csv'
+        fname = f"{self.user}_{namespace}.csv"
         if self.path != '/':
-            f_name = self.path + f_name
-
-        if f_name in os.listdir(self.path):
-            store_value.to_csv(f_name, sep=',', header=False,
-                               mode='a', index=False)
+            fname = f"{self.path}{fname}"
```
Comment (hdiogenes) [inline backends/csvbackend.py]: Use `os.path.join` or `pathlib`

File: backends/csvbackend.py [hunk 4]
```diff
@@ -68,52 +64,60 @@ def _read_config(self, settings):
...
+        """ Delete a instances of the csv file."""
+
+        fname = f"{self.user}_{namespace}.csv"
         if self.path != '/':
-            file = self.path + file
-
-        dataframe.to_csv(file, sep=',', header=True, mode='w', index=False)
-
-    def get(self, file, start=None, end=None, method=None,
+            fname = f"{self.path}{fname}"
```
Comment (hdiogenes) [inline backends/csvbackend.py]: os.path.join

File: tests/__init__.py [hunk 5]
```diff
@@ -0,0 +1,13 @@
+"""Module to test the napp kytos/kronos."""
+import os
+import sys
+from pathlib import Path
+
+if 'VIRTUAL_ENV' in os.environ:
+    BASE_ENV = Path(os.environ['VIRTUAL_ENV'])
+else:
+    BASE_ENV = Path('/')
```
Comment (hdiogenes) [inline tests/__init__.py]: `BASE_ENV = Path(os.environ.get('VIRTUAL_ENV', '/'))`

File: tests/test_csvbackend.py [hunk 6]
```diff
@@ -0,0 +1,56 @@
+"""Module to test the main napp file."""
```
Comment (hdiogenes) [inline tests/test_csvbackend.py]: update docstring

File: tests/test_csvbackend.py [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+import time
+from unittest import TestCase
+
+from napps.kytos.kronos import settings
+from napps.kytos.kronos.backends.csvbackend import CSVBackend
+from napps.kytos.kronos.utils import now
+
+
+class Test_CSVBackend(TestCase):
```
Comment (hdiogenes) [inline tests/test_csvbackend.py]: remove underscore from class name

File: tests/test_csvbackend.py [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+    def test_save_success(self):
+        """ Test to check the success in data storage in CSV Backend."""
+        namespace = 'kytos.telemetry.switches.1.interfaces.232.bytes_in'
+        value = '1324'
+        timestamp = now()
+
+        csvbackend = CSVBackend(settings)
+        result = csvbackend.save(namespace, value, timestamp)
+
+        self.assertEqual(True, result)
```
Comment (hdiogenes) [inline tests/test_csvbackend.py]: No need to return True. None is enough.

File: backends/csvbackend.py [hunk 9]
```diff
@@ -68,52 +60,53 @@ def _read_config(self, settings):
...
-        """Retrieve data from a csv file"""
-        dataframe, file = self._load_file(file)
-        search = _make_search(start, end, dataframe)
+        rows = []
+
+        fname = f"{self.user}_{namespace}.csv"
+        fname = str(Path(self.path, fname))
+ 
+        if not os.path.exists(self.path):
+            rows.append(['Value', 'Timestamp'])
```
Comment (hdiogenes) [inline backends/csvbackend.py]: Shouldn't you check for `fname` instead of `self.path`?

File: backends/csvbackend.py [hunk 10]
```diff
@@ -68,52 +60,53 @@ def _read_config(self, settings):
...
-
-    def get(self, file, start=None, end=None, method=None,
-            fill=None, group=None):
-        """Retrieve data from a csv file"""
-        dataframe, file = self._load_file(file)
-        search = _make_search(start, end, dataframe)
+        rows = []
+
+        fname = f"{self.user}_{namespace}.csv"
+        fname = str(Path(self.path, fname))
```
Comment (hdiogenes) [inline backends/csvbackend.py]: Do you really need to convert it to `str`?

File: backends/csvbackend.py (hunk 9 above)
Comment (josemauro) [reply backends/csvbackend.py]: Yes.

File: backends/csvbackend.py (hunk 10 above)
Comment (josemauro) [reply backends/csvbackend.py]: Yes.

**Pull request merged**